### PR TITLE
Fix bug and also lsp js support

### PR DESCRIPTION
**Pull request by ajbt200128**
This PR fixes a bug where the language server wouldn't run on save, #33, and also adds support for lspjs

Apologies this is all one PR, but a bunch of this stuff was related and I don't have the willpower to separate it all out

This PR fixes a bug wherein when a file was changed, the Language Server was not notified. It also refactors SemgrepLspServer to not depend on LspServer going forwards.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -122,3 +140,7 @@ tasks {
             properties("pluginVersion").map { listOf(it.split('-').getOrElse(1) { "default" }.split('.').first()) }
     }
 }
+
+fun downloadLspJs() {
+
+}
```
Comment (tpetr) [inline build.gradle.kts:146]: missing impl?

File: src/main/kotlin/com/semgrep/idea/settings/LspSettings.kt [hunk 2]
```diff
@@ -18,6 +18,9 @@ data class SemgrepLspSettings(
     var doHover: Boolean = false,
     var metrics: Metrics = Metrics(),
     var pro_intrafile: Boolean = false,
+    var useJS: Boolean = System.getProperty("os.name").lowercase(Locale.getDefault()).contains("win"),
+    var stackSizeJS: Int = 1000 * 1000,
```
Comment (tpetr) [inline src/main/kotlin/com/semgrep/idea/settings/LspSettings.kt:22]:
```suggestion
    var stackSizeJS: Int = 1024 * 1024,
```

**Review by tpetr — COMMENTED** [review]
was the `lspjs/dist` commit intentional?

File: build.gradle.kts (hunk 1 above)
Comment (ajbt200128) [reply build.gradle.kts:146]: No, just accidentally left over from trying something, will remove

**Review by tpetr — APPROVED** [review]
LGTM if tested

**Comment by ajbt200128** [comment]
It's tested and working for me :D

**Pull request merged**